ompi/mca/common/ompio/common_ompio_buffer.c

```diff
@@ -66,14 +63,18 @@ static void* mca_common_ompio_buffer_alloc_seg ( void*ctx, size_t *size )
 {
     char *buf=NULL;
     size_t realsize, numpages;
+    int tmp = -1;
+    uint64_t flags = -1;
 
     numpages = (*size + mca_common_ompio_pagesize -1 )/mca_common_ompio_pagesize;
     realsize = numpages * mca_common_ompio_pagesize;
 
     buf = malloc ( realsize);
 
-    if (NULL != buf && NULL != opal_accelerator.host_register) {
-        opal_accelerator.host_register((void *)buf, realsize);
+    if (NULL != buf) {
+        if (opal_accelerator.check_addr(buf, &tmp, &flags) > 0) {
```

Comment:
Pretty sure this check is wrong, see:

https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__MEMORY.html#group__CUDART__MEMORY_1ge8d5c17670f16ac4fc8fcb4181cb490c

Reply:
removed commit